lib/schedule.c

```diff
@@ -142,7 +86,15 @@ trigger_handle_t lf_schedule_value(void* action, interval_t extra_delay, void* v
         );
         return -1;
     }
-    return _lf_schedule_value((lf_action_base_t*)action, extra_delay, value, (size_t)length);
+    token_template_t* template = (token_template_t*)action;
+    environment_t* env = action->parent->environment;
+    LF_CRITICAL_SECTION_ENTER(env);
+    lf_token_t* token = _lf_initialize_token_with_value(template, value, length);
+    int return_value = lf_schedule_trigger(env, action->trigger, extra_delay, token);
+    // Notify the main thread in case it is waiting for physical time to elapse.
+    lf_notify_of_event(env);
+    LF_CRITICAL_SECTION_EXIT(env);
+    return return_value;
```

Comment:
Note to other reviewers: this part is the same as the original implementation of `_lf_schedule_value`